includes/express-checkout/class-wc-payments-express-checkout-button-helper.php

```diff
@@ -1033,6 +1053,39 @@ public function update_shipping_method( $shipping_methods ) {
 		WC()->session->set( 'chosen_shipping_methods', $chosen_shipping_methods );
 	}
 
+	/**
+	 * Add express checkout payment method title to the order.
+	 *
+	 * @param integer $order_id The order ID.
+	 *
+	 * @return  void
+	 */
+	public function add_order_payment_method_title( $order_id ) {
+		if ( empty( $_POST['express_payment_type'] ) || ! isset( $_POST['payment_method'] ) || 'woocommerce_payments' !== $_POST['payment_method'] ) { // phpcs:ignore WordPress.Security.NonceVerification
+			return;
+		}
+
+		$express_payment_type   = wc_clean( wp_unslash( $_POST['express_payment_type'] ) ); // phpcs:ignore WordPress.Security.NonceVerification
+		$express_payment_titles = [
+			'apple_pay'  => 'Apple Pay',
+			'google_pay' => 'Google Pay',
+		];
+		$payment_method_title   = $express_payment_titles[ $express_payment_type ] ?? false;
```

Comment:
@cesarcosta99 In `add_order_meta` there was a fallback title "Express Payment", which I believe was an adaptation of the legacy "Payment Request" title ([src](https://github.com/Automattic/woocommerce-payments/blob/78fba818fc3462ce4f475041a03cff0256ce2cad/includes/class-wc-payments-payment-request-button-handler.php#L752)), but I've removed it since Stripe no longer supports the fallback purple "Payment Request" button and other wallets should be predefined here.

Reply:
I confirmed the fallback when this returns early is `Credit card / debit card` which is sufficient for a case where we couldn't determine the title successfully. 👍